src/modules/licences/controllers/agreements.js

```diff
@@ -19,5 +21,36 @@ const getLicenceAgreements = async request => {
   return licencesService.getLicenceAgreementsByLicenceRef(licence.licenceNumber);
 };
 
+const deleteAgreement = async (request, h) => {
+  const { agreementId } = request.params;
+  const { email: issuer } = request.defra.internalCallingUser;
+  try {
+    // Get licence agreement to save in event metadata
+    const licenceAgreement = await licencesService.getLicenceAgreementById(agreementId);
+
+    if (!licenceAgreement) {
+      return Boom.notFound(`Agreement ${agreementId} not found`);
+    }
+
+    // Delete the licence agreement
+    await licencesService.deleteLicenceAgreementById(agreementId);
+
+    // log deletion in event log
+    await eventHelper.saveEvent(
```

Comment:
I wonder if the event logging code could also go in the service - however I'm happy to move this over later.